src/services/clients/cxClient.ts

```diff
@@ -319,20 +319,54 @@ export class CxClient {
             this.isNewProject = false;
         } else {
             this.log.info('Project not found, creating a new one.');
-            this.isNewProject = true;
-            if (this.sastConfig.denyProject) {
+            if (this.sastConfig.denyProject) 
                 throw Error(
                     `Creation of the new project [${this.config.projectName}] is not authorized. Please use an existing project.` +
                     " You can enable the creation of new projects by disabling the Deny new Checkmarx projects creation checkbox in the Checkmarx plugin global settings.");
+            
+            //Create newProject and checking if EnableSASTBranching is enabled then creating branch project 
+            if (this.sastConfig.enableSastBranching) 
+            {
+                if(!this.sastConfig.masterBranchProjectName)
+                    throw Error("Master branch project name is mandatory to create branched project.");
+                else
+                {
+                    let masterProjectId = await this.getProjectIdByProjectName(this.sastConfig.masterBranchProjectName);
+
+                    if(masterProjectId)
+                    {
+                        projectId = await this.createChildProject(masterProjectId,this.config.projectName);
+                        if(projectId == -1 )
+                            throw Error('Branched project could not be created: ' + this.config.projectName);
```

Comment:
Add the curly brackets to if condition

Reply:
In a if condition single statement is there so not required. As a best practice reduces the no. of line of code.